gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/sideEffect/AggregateGlobalStep.java

```diff
@@ -122,8 +121,16 @@ protected Traverser.Admin<S> processNextStart() {
 
     @Override
     public void processAllStarts() {
+        final TraversalSideEffects sideEffects = this.getTraversal().getSideEffects();
+
+        // Pre-defined Operator such as addAll and assign will reduce over the whole input set, rather than
+        // applying a single input one by one.
+        final boolean isOperatorForBulkSet = sideEffects.getReducer(sideEffectKey) == Operator.addAll ||
```

Comment:
nit: suggest renaming this to `bulkReducingOperator` or something similar to callout observation towards the nature of reducing operators instead of limiting to BulkSet. addAll works with either maps or any collection and assign works with any type of input.

Reply:
thanks, renamed it